p2p/p2p/src/peer_set/client_wrappers.rs

```diff
@@ -0,0 +1,86 @@
+use std::{
+    ops::{Deref, DerefMut},
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+};
+
+use cuprate_p2p_core::{
+    client::{Client, WeakClient},
+    NetworkZone,
+};
+
+/// A client stored in the peer-set.
+pub(super) struct StoredClient<N: NetworkZone> {
+    pub client: Client<N>,
+    /// An [`AtomicBool`] for if the peer is currently downloading blocks.
+    downloading_blocks: Arc<AtomicBool>,
+    /// An [`AtomicBool`] for if the peer is currently being used to stem txs.
+    stem_peer: Arc<AtomicBool>,
```

Comment:
To keep track of peers that are already downloading blocks/stemming txs we use atomic bools